### PR TITLE
Added documentation required by issues #264, #284 and #676

### LINKED ISSUE #264: document module path handling when loading modules - how modules are resolved

### PR DESCRIPTION
**Pull request by omusil24**
Fixes #264, #284, #676

### CODE REVIEW

File: doxygen/lang/120_modules.dox.tmpl [hunk 1]
```diff
@@ -6,7 +6,18 @@
 
     %Qore modules allow the %Qore language to be extended at run-time.
 
-    To load a module at parse time (normally required for most modules), use the @ref requires "%requires" or @ref try-module "%try-module" parse directive. If the named feature is not already present in %Qore, %Qore looks for a module with this name in the directories listed in the \c QORE_MODULE_DIR environment variable (see @ref environment_variables).
+    To load a module at parse time (normally required for most modules), use the @ref requires "%requires" or @ref try-module "%try-module" parse directive. If the named feature is not already present in %Qore, %Qore will try to load the module from it's list of module directories (module path). This list contains the following directories:\n
```
Comment (davidnich) [inline doxygen/lang/120_modules.dox.tmpl]: `it's` should be `its` (possessive pronoun) - `it's` = `it is`

File: doxygen/lang/120_modules.dox.tmpl [hunk 2]
```diff
@@ -6,7 +6,18 @@
...
+    -# directories listed in the environment variable \c QORE_MODULE_DIR (see @ref environment_variables)
+    -# USER_MODULE_VER_DIR - build variable set to directory where Qore user modules of the specific Qore version should be installed (ex: \c /usr/share/qore-modules/0.8.12)
```
Comment (davidnich) [inline doxygen/lang/120_modules.dox.tmpl]: `# \c USER_MODULE_VER_DIR`
etc

**Review by davidnich — CHANGES_REQUESTED** [review]
minor comments

**Comment by omusil24** [comment]
Fixed ;)

**Pull request merged**